internal/core/src/mmap/Column.h

```diff
@@ -944,22 +956,26 @@ class SingleChunkArrayColumn : public SingleChunkColumnBase {
     void
     ConstructViews() {
         views_.reserve(indices_.size());
-        for (size_t i = 0; i < indices_.size() - 1; i++) {
+        auto last = indices_.size() - 1;
+        for (size_t i = 0; i < last; i++) {
             views_.emplace_back(data_ + indices_[i],
+                                lens_[i],
                                 indices_[i + 1] - indices_[i],
                                 element_type_,
-                                std::move(element_indices_[i]));
+                                element_indices_[i].data());
         }
         views_.emplace_back(data_ + indices_.back(),
+                            lens_[last],
                             data_size_ - indices_.back(),
                             element_type_,
-                            std::move(element_indices_[indices_.size() - 1]));
-        element_indices_.clear();
+                            element_indices_[last].data());
+        lens_.clear();
     }
```

Comment:
we cannot clear element_indices_ anymore, because array_view just hold the pointer of these vectors instead of owning these data in memory